### PR TITLE
Integration tests: add implicit gomod test

### PR DESCRIPTION
**Pull request by ejegrova**
Added test that uses only repo and ref as payload for request- Package manager is implicitly gomod.

### CODE REVIEW

File: tests/integration/test_packages.py [hunk 1]
```diff
@@ -34,14 +35,21 @@ def test_packages(env_package, env_name, test_env, tmpdir):
...
+    if env_name == "implicit_gomod":
+        payload = {
+            "repo": env_data["repo"],
+            "ref": env_data["ref"],
+        }
+    else:
+        payload = {
             "repo": env_data["repo"],
             "ref": env_data["ref"],
             "pkg_managers": env_data["pkg_managers"],
```
Comment (athos-ribeiro) [inline tests/integration/test_packages.py:48]: How about just changing this line to 

`"pkg_managers": env_data.get("pkg_managers", []),`

and adding the following after the payload assignment:

```
if env_name == "implicit_gomod":
    payload.pop(pkg_managers)
```

File: tests/integration/test_packages.py [hunk 2]
```diff
@@ -34,14 +35,21 @@ def test_packages(env_package, env_name, test_env, tmpdir):
...
             "flags": env_data.get("flags", []),
```
Comment (athos-ribeiro) [inline tests/integration/test_packages.py:49]: It would also be nice to change this one (in a different commit, could be a different PR.
We are always testing the request with flags. we should just add it later if it exists, like
```
if env_data.get("flags"):
    payload["flags"] = env_data.get["flags"]
```

and then adding a single test passing empty flags just to check Cachito's sanity.

**Review by athos-ribeiro — COMMENTED** [review]
LGTM!

Just a small nitpick. Feel free to just ignore the comment about the `flags` we can file an issue for that later :)

File: tests/integration/test_packages.py (hunk 1 above)
Comment (ejegrova) [reply tests/integration/test_packages.py:48]: oh, I spend some time today trying to figure out how to simplify this but I didn't think of pop :D

File: tests/integration/test_data/gomod_packages.yaml [hunk 3]
```diff
@@ -202,3 +202,86 @@ vendored_without_flag:
   repo: https://github.com/cachito-testing/gomod-vendored.git
   ref:  ff1960095dd158d3d2a4f31d15b244c24930248b
   pkg_managers: ["gomod"]
+implicit_gomod:
```
Comment (akhmanova) [inline tests/integration/test_data/gomod_packages.yaml:205]: Can you please add a comment about the test data?

File: tests/integration/test_packages.py (hunk 2 above)
Comment (akhmanova) [reply tests/integration/test_packages.py:43]: Yes, a good idea.
Or like this one: 
```suggestion
           if "flags" in env_data:
               payload["flags"] = env_data["flags"]
```

Comment (ejegrova) [reply tests/integration/test_packages.py:43]: I think we should file new issue for that and add new test to the testing doc.

File: tests/integration/test_data/gomod_packages.yaml (hunk 3 above)
Comment (ejegrova) [reply tests/integration/test_data/gomod_packages.yaml:206]: done :)

**Review by athos-ribeiro — CHANGES_REQUESTED** [review]
Thanks for the changes!

There is a lint error now. Could you fix that?

**Review by athos-ribeiro — APPROVED** [review]

**Comment by akhmanova** [comment]
@ejegrova Looks good, what is the purpose of empty commit?

**Comment by ejegrova** [comment]
> @ejegrova Looks good, what is the purpose of empty commit?

sorry about that, wrong branch for testing Jenkins, it's removed now :)

**Review by athos-ribeiro — APPROVED** [review]
Good catch on the empty commit, @akhmanova
